ccl/backupccl: TestGCDropIndexSpanExpansion failed

ccl/backupccl.TestGCDropIndexSpanExpansion [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4162346&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4162346&tab=artifacts#/) on master @ [46bdd4e299b3324eb65b943dc4a1c10ded7d8a7a](https://github.com/cockroachdb/cockroach/commits/46bdd4e299b3324eb65b943dc4a1c10ded7d8a7a):


```
=== RUN   TestGCDropIndexSpanExpansion
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestGCDropIndexSpanExpansion3544265402
    test_log_scope.go:80: use -show-logs to present logs inline
    backup_test.go:9036: error executing 'BACKUP INTO LATEST IN 'nodelocal://0/foo' WITH revision_history': pq: failed to run backup: exporting 13 ranges: exporting /Table/56/{3-4}: batch timestamp 1642664115.146851995,0 must be after replica GC threshold 1642664116.550345007,0
    panic.go:661: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestGCDropIndexSpanExpansion3544265402
--- FAIL: TestGCDropIndexSpanExpansion (9.85s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/bulk-io
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGCDropIndexSpanExpansion.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestGCDropIndexSpanExpansion [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4169174&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4169174&tab=artifacts#/) on master @ [ebda0ecb4aa1fe47f1403635846e342a2cfbfa1b](https://github.com/cockroachdb/cockroach/commits/ebda0ecb4aa1fe47f1403635846e342a2cfbfa1b):


```
=== RUN   TestGCDropIndexSpanExpansion
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestGCDropIndexSpanExpansion3501611901
    test_log_scope.go:80: use -show-logs to present logs inline
    backup_test.go:9036: error executing 'BACKUP INTO LATEST IN 'nodelocal://0/foo' WITH revision_history': pq: failed to run backup: exporting 13 ranges: exporting /Table/56/{1-2}: batch timestamp 1642719965.243485028,0 must be after replica GC threshold 1642719966.582763509,0
    panic.go:661: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestGCDropIndexSpanExpansion3501611901
--- FAIL: TestGCDropIndexSpanExpansion (9.38s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGCDropIndexSpanExpansion.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestGCDropIndexSpanExpansion [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4175440&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4175440&tab=artifacts#/) on master @ [72655fd4730fadb0ddacd8a5513a5d2d9121cbc6](https://github.com/cockroachdb/cockroach/commits/72655fd4730fadb0ddacd8a5513a5d2d9121cbc6):


```
=== RUN   TestGCDropIndexSpanExpansion
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestGCDropIndexSpanExpansion1354184770
    test_log_scope.go:80: use -show-logs to present logs inline
    backup_test.go:9036: error executing 'BACKUP INTO LATEST IN 'nodelocal://0/foo' WITH revision_history': pq: failed to run backup: exporting 13 ranges: exporting /Table/56/{1-2}: batch timestamp 1642777676.763932686,0 must be after replica GC threshold 1642777678.034359492,0
    panic.go:661: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestGCDropIndexSpanExpansion1354184770
--- FAIL: TestGCDropIndexSpanExpansion (9.51s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGCDropIndexSpanExpansion.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestGCDropIndexSpanExpansion [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4197056&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4197056&tab=artifacts#/) on master @ [421a767fa41a67efa157a96be8b8e43cdbc55b63](https://github.com/cockroachdb/cockroach/commits/421a767fa41a67efa157a96be8b8e43cdbc55b63):


```
=== RUN   TestGCDropIndexSpanExpansion
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestGCDropIndexSpanExpansion1403063623
    test_log_scope.go:80: use -show-logs to present logs inline
    backup_test.go:9036: error executing 'BACKUP INTO LATEST IN 'nodelocal://0/foo' WITH revision_history': pq: failed to run backup: exporting 13 ranges: exporting /Table/56/{1-2}: batch timestamp 1643070671.926476161,0 must be after replica GC threshold 1643070673.308443057,0
    panic.go:661: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestGCDropIndexSpanExpansion1403063623
--- FAIL: TestGCDropIndexSpanExpansion (9.45s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGCDropIndexSpanExpansion.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/backupccl/backup_test.go
+++ b/pkg/ccl/backupccl/backup_test.go
@@ -9007,11 +9007,12 @@ func TestGCDropIndexSpanExpansion(t *testing.T) {
 	}})
 	defer tc.Stopper().Stop(ctx)
 	sqlRunner := sqlutils.MakeSQLRunner(tc.Conns[0])
+	sqlRunner.Exec(t, `SET CLUSTER SETTING kv.closed_timestamp.target_duration = '100ms'`) // speeds up the test
 
 	sqlRunner.Exec(t, `
 CREATE DATABASE test; USE test;
 CREATE TABLE foo (id INT PRIMARY KEY, id2 INT, id3 INT, INDEX bar (id2), INDEX baz(id3));
-ALTER TABLE foo CONFIGURE ZONE USING gc.ttlseconds = '1';
+ALTER INDEX foo@bar CONFIGURE ZONE USING gc.ttlseconds = '1';
 INSERT INTO foo VALUES (1, 2, 3);
 DROP INDEX foo@bar;
 `)
@@ -9033,6 +9034,25 @@ DROP INDEX foo@bar;
 	// Wait for the GC to complete.
 	jobutils.WaitForJob(t, sqlRunner, gcJobID)
 
+	waitForTableSplit := func() {
+		testutils.SucceedsSoon(t, func() error {
+			count := 0
+			sqlRunner.QueryRow(t,
+				"SELECT count(*) "+
+					"FROM crdb_internal.ranges_no_leases "+
+					"WHERE table_name = $1 "+
+					"AND database_name = $2",
+				"foo", "test").Scan(&count)
+			if count == 0 {
+				return errors.New("waiting for table split")
+			}
+			return nil
+		})
+	}
+	waitForTableSplit()
+
+	// This backup should succeed since the spans being backed up have a default
+	// GC TTL of 25 hours.
 	sqlRunner.Exec(t, `BACKUP INTO LATEST IN 'nodelocal://0/foo' WITH revision_history`)
 }
 

